### PR TITLE
Refactor Faction GUI system

### PR DESCRIPTION
**Pull request by dariasc**
Rewrite the old ugly UI system, now contains:

- Dynamic items that change depending on User state
- Cleaner and less duplicate code
- Users can define base items that ui components will be merged on
- Dummy and base items are now built on server start, reducing a little load

### CODE REVIEW

**Comment by dariasc** [comment]
I still need to do some testing with the changes I've made.
Would love any more ideas to clean this up a bit more.

Might be a good idea to JavaDoc this a bit, was a bit hard for me to wrap my head around the changes I made

**Comment by dariasc** [comment]
Implemented a Back system currently only in use by Action.

Updated config to use new `Materials.JSON` data

perms

File: src/main/java/com/massivecraft/factions/zcore/ui/FactionUI.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        }
+
+        buildDummyItems();
+        buildItems();
+    }
+
+    protected void buildItems() {
+        ConfigurationSection items = config.getConfigurationSection("items");
+        if (items == null) {
+            return;
```
Comment (drtshock) [inline src/main/java/com/massivecraft/factions/zcore/ui/FactionUI.java]: Log an error here?

File: src/main/java/com/massivecraft/factions/zcore/ui/FactionUI.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            item = parse(item, type);
+
+            inventory.setItem(entry.getKey(), item.get());
+        }
+    }
+
+    protected void buildDummyItems() {
+        ConfigurationSection dummies = config.getConfigurationSection("dummies");
+        if (dummies == null) {
+            return;
```
Comment (drtshock) [inline src/main/java/com/massivecraft/factions/zcore/ui/FactionUI.java]: Also log an error here if we're unable to build a GUI.

**Review by drtshock — CHANGES_REQUESTED** [review]

**Review by drtshock — APPROVED** [review]

**Pull request merged**